lib/mpool/include/mpool/mpool_structs.h

```diff
@@ -57,17 +58,20 @@ struct mpool_cparams {
         uint8_t  filecnt;
         char     path[PATH_MAX];
     } mclass[MP_MED_COUNT];
+    bool pmem_only;
 };
 
 /**
  * struct mpool_rparams - mpool run params
  *
- * @path: storage path
+ * @path:      storage path
+ * @pmem_only: pmem-only KVDB
  */
 struct mpool_rparams {
     struct {
         char path[PATH_MAX];
     } mclass[MP_MED_COUNT];
+    bool pmem_only;
```

Comment:
Not a huge fan of this member? Can't mpool determine internally whether it is pmem-only? Would be a pretty quick check.

Reply:
I think it will nicer if mpool is completely unware of what "pmem-only" is.

The "pmem_only" mpool c/r param is currently used within mpool only for some path checks. Let me see if I can move these checks to the ikvdb layer and get rid of this newly added param.

Reply:
Done, with the latest commit mpool is unaware of what a "pmem-only" KVDB is.